### DISCUSSION TITLE
Responsive design with grid columns not working properly(?)

### DISCUSSION THREAD

**Discussion by LivingInLimbo** [tailwindlabs/tailwindcss#8272] [Help] [answered]
The class: `flex justify-center w-full grid auto-rows-fr grid-cols-2 sm:grid-cols-3 lg:grid-cols-4 2xl:grid-cols-6`

When I use this class, the grid always stays at two columns no matter what size the screen is. However, if I change the class to `flex justify-center w-full grid auto-rows-fr grid-cols-1  sm:grid-cols-2 lg:grid-cols-4 2xl:grid-cols-6`, the grid resizes as expected from one to two columns, but not from two columns to four columns at the large breakpoint.

Essentially I can only get the grid columns to resize on a breakpoint if they initially start at one column, and resize in one column increments.

This worked fine before upgrading to version 3, so I'm wondering if anyone has the same issue. I didn't see any changes to grid columns in the upgrade guide.

Edit: After some tinkering I'm pretty sure it's to do with how I'm generating the classnames, basically they're a string that's generated upon render. I'm still not sure why some of the classes work but others don't.

**Comment by LivingInLimbo** [comment, marked as answer]
After some tinkering I'm pretty sure it's to do with how I'm generating the classnames, basically they're a string that's generated upon render. I'm still not sure why some of the classes work but others don't.

**Reply by sachinthaHeshan** [reply]
Don't put class names like this : className={`grid xl:grid-cols-${isOpen ? '3' : '1'} grid-cols-1 gap-6`}
Right way : className={`grid ${isOpen ? 'xl:grid-cols-3' : 'xl:grid-cols-1'} grid-cols-1 gap-6`}

**Reply by inbarindieflow** [reply]
This solution works great. 
Theres a reason why some classes work dynamically and others don't. 
The classes that work dynamically, you have probably written hard-coded before so they are already statically detectable at build-time, and appears as exist as complete unbroken strings in your source files.

https://tailwindcss.com/docs/content-configuration#dynamic-class-names
